netcore/System.Private.CoreLib/src/System/Environment.Unix.cs

```diff
@@ -13,36 +13,91 @@ namespace System
 {
 	partial class Environment
 	{
+		private static Dictionary<string, string> s_environment;
+
 		static string GetEnvironmentVariableCore (string variable)
 		{
-			using (var h = RuntimeMarshal.MarshalString (variable)) {
-				return internalGetEnvironmentVariable_native (h.Value);
+			Debug.Assert(variable != null);
+
+			if (s_environment == null) {
+				using (var h = RuntimeMarshal.MarshalString (variable)) {
+					return internalGetEnvironmentVariable_native (h.Value);
+				}
+			}
+
+			lock (s_environment) {
```

Comment:
Are you missing `EnsureEnvironmentCached ();` call?

Reply:
Not missing, `GetEnvironmentVariableCore` has a fast path for `s_environment == null`.